shared/.js/src/main/scala/urldsl/url/JSUrlStringParser.scala

```diff
@@ -24,7 +24,8 @@ final class JSUrlStringParser(val rawUrl: String) extends UrlStringParser {
 
 object JSUrlStringParser {
 
-  final lazy val jsUrlStringParserGenerator: UrlStringParserGenerator =
-    (rawUrl: String) => new JSUrlStringParser(rawUrl)
+  final lazy val jsUrlStringParserGenerator: UrlStringParserGenerator = new UrlStringParserGenerator {
+    def parser(rawUrl: String): UrlStringParser = new JSUrlStringParser(rawUrl)
+  }
```

Comment:
Made this more explicit because trait usages in SAM syntax are really hard to find, especially when IntelliJ is botched from all the platform-specific code